### DISCUSSION TITLE
Bug: Break thru protected blocks

### DISCUSSION THREAD

**Discussion by ShaneBeee** [GriefPrevention/GriefPrevention#198] [Ideas]
### What behaviour is expected:
That you can not break blocks on another persons claim.
### What behaviour is observed:
Holding down both right and left click, you are able to break blocks momentarily, and open chests
### Steps/models to reproduce:
Start by holding down right click, and then proceed to holding down left click. Doing this breaks the block temporarily and you are able to open a chest that is behind the block.
PS we have chest protection turned off on our server, so that siege mode gives a little more advantage to the winner. But the problem, is, a player is able to (without being in siege mode) open my chests from behind a wall. 
### Spigot/Craftbukkit/Paper and GriefPrevention version:
CraftBukkit version git-Spigot-65e8124-357b573 (MC: 1.12.1) (Implementing API version 1.12.1-R0.1-SNAPSHOT)
GriefPrevention version 16.7.1

**Comment by bigpresh** [comment]
I rather suspect this is just a case of "this is how Minecraft works" - if you've disabled chest protection, then Grief Protection won't protect chests, because you asked it not to.  It will cancel block break events, but, especially if there's a little lag or network latency involved, that can take a brief moment; if the client meanwhile accesses the chest, GP won't stop them, because you asked it no to.  (I suspect you'd find someone using a hacked client to see through the wall, e.g. some form of freecam mode, would also be able to access the chests.)

**Comment by RoboMWM** [comment]
bigpresh is right on the money.

As you may already be aware, MC doesn't do too much/care about client hacks. Therefore it doesn't do any LOS check when the client requests to open the inventory of a block. You might have some luck if you install NoCheatPlus or a similar anticheat solution, but the only way to guarantee protection is to enable chest protection.

An addon _may_ be able to allow chest access during siege. GP version 20 will provide better API access for such addons.

**Comment by bigpresh** [comment]
Gut feeling is that the best thing to give you what you want would be a plugin that extends GP, hooking in to the container access checks and allowing it if the container's owner just lost a siege in that claim and the person trying to access it was the attacker - but I'm pretty sure GP as it is doesn't provide a clean way to hook in to add that.  That way containers would ordinarily be safely protected, but after winning a siege, the attacker could have a time window to plunder the defender's stuff.

**Comment by ShaneBeee** [comment]
I tried it myself, its not a hack, its just straight up MC client!
But After what you said, it makes sense its not a GP problem..... just was hoping it was something that could be fixed!

I wish someone would properly create a siege add on, because the built in one is terrible! :(

**Comment by RoboMWM** [comment]
> I tried it myself, its not a hack, its just straight up MC client!

While a hacked client can exploit this, I never mentioned that this was the result of a hacked client; I stated the server doesn't do a LOS (line-of-sight) check. The client thinks it is truly interacting with the chest behind the block, because as far as **the client** is concerned, the block is broken (and, more importantly, the vanilla client _does not wait for the server to confirm_ this). Since the server doesn't confirm if the block is indeed in the line of sight of the player, it delivers the block's inventory.

Plugins like NCP will perform this line-of-sight check and cancel the interaction altogether if it fails, preventing the server from delivering the inventory.
